### ISSUE TITLE
Suggestion for blank new tab urls, use full page iframes

### ISSUE THREAD

**Issue by benjaminjackman** [cadeyrn/newtaboverride#141]
So I just read the commentary over at https://bugzilla.mozilla.org/show_bug.cgi?id=1409675 ... and uh yikes. 

So I've written extensions for newtab pages for personal use for both firefox and chrome that have had blank urls when opening new tabs while still allowing me to specify content dynamically.

My trick was to specify a file in the manifest that acted as a wrapper that had an iframe that scaled up to the entire page and which then loaded whatever real page I wanted:

manifest.json
```json
...
  "chrome_url_overrides": {
    "newtab": "newtab-outer.html"
  }
}
```

The wrapper: newtab-outer.html
```html
<html>
<head>
  <title>New Tab</title>
  <style>
    * {
      margin: 0;
      padding: 0;
    }

    html, body {
      height: 100%;
      width: 100%;
      overflow: hidden;
    }

    table {
      height: 100%;
      width: 100%;
      table-layout: static;
      border-collapse: collapse;
    }

    iframe {
      height: 100%;
      width: 100%;
    }

    .header {
      border-bottom: 1px solid #000;
    }

    .content {
      height: 100%;
    }
  </style>
</head>
<body>
<!-- This lovely pile of hacks is done so that the internal iframe can be loaded dynamically,
while leaving the address bar empty -->
<iframe src="newtab/newtab.html" frameborder="0"></iframe>
<!-- if needed to load dynamically a script could be used here to set the iframe url etc-->
<!--<script src="load-inner-tab.js"></script>-->
</body>

</html>
```

When specified in this way firefox / chrome will not fill in the url.

The only downside to this (other than being a pile of hacks) is that whatever page is in the wrapper has to use target="_parent" to get the links to correctly escape the wrapper iframe, as I custom wrote that inner page (it's a handwritten home page I can share between mobile and desktop) it wasn't a problem for me.

**Comment by cadeyrn** [comment]
Hi,

thank you for posting your workaround. It's no longer necessary because there is a visible option in about:preferences#home since Firefox 61. That's why I removed the option for a blank new tab in the latest update of New Tab Override.

**Comment by benjaminjackman** [comment]
Well this wasn't really for a blank new tab, it was more about a way of leaving the address bar blank while showing user controlled content in the page.

**Comment by cadeyrn** [comment]
This approach can't be used for all websites which forbid the embedding via iframe, for example Google. ;)

**Comment by benjaminjackman** [comment]
Well, I'm just using it for my own homepage, not embed google or anything
and figured you might find it a useful idea for your users.

As a side note, it is possible to get google to load in an iframe by
writing a browser extension (at least in chrome) to intercept and modify the
X-FRAME-OPTIONS / CSP headers. I did this for a side project I never
finished. Some sites (like ESPN for example used to be this way)
have additional overrides where they use javascript to detect they are in
an iframe, but when I looked the google sites did not.

Anyhow if you cared about getting something like google to load in an
iframe you might be able to get it to work by merging
those two approaches in your extension.

Is it worth the trouble? i dunno :D
